katas/content/superposition/parity_bitstrings/solution.md

```diff
@@ -0,0 +1,56 @@
+
+1. Apply **H** gate to the first qubit to prepare state $\frac{1}{\sqrt{2}} ( |0\rangle + |1\rangle ) \otimes |0\rangle_{N-1}$
+2. Apply the controlled variant of procedure of preparing $|P_{N-1,p}\rangle$ on the last $N-1$ qubits, with the first qubit in the $|0\rangle$ state as the control.
+3. Apply the controlled variant of procedure of preparing $|P_{N-1,1-p}\rangle$ on the last $N-1$ qubits, with the first qubit in the $|1\rangle$ state as the control.
+
+> Q# library function [`ApplyControlledOnInt`](https://learn.microsoft.com/qsharp/api/qsharp-lang/microsoft.quantum.canon/applycontrolledonint) allows to do that easily.
+
+4. The base of recursion is preparing the states for $N = 1$:
+* For `parity = 0`, there is one single-qubit state with this parity: $|0\rangle$ (no action required to prepare).
+* For `parity = 1`, there is one single-qubit state with this parity: $|1\rangle$ (apply **X** gate to prepare).
+
+@[solution]({ "id": "superposition__parity_bitstrings_solution_a", "codePath": "./SolutionA.qs" })
+
+We start by preparing an equal superposition of all basis states and allocating an extra qubit.
+
+This time we use the extra qubit to calculate the parity of the input state: applying a series of CNOT gates, each one with one of the input qubits as control and the extra qubit as a target will compute the parity of the state.
+
+Now we measure the extra qubit: if the measurement result matches our parity, we're done — the input qubits collapsed to an equal superposition of all states that have this parity. If the measurement result is the opposite, we can retry the whole process.
+
+We can avoid retrying the state preparation if our measurement result doesn't match the required parity: notice that applying an X gate to any one of the qubits changes the parity of each basis state to the opposite one, and thus converts the state we got to the state we need.
+
+@[solution]({ "id": "superposition__parity_bitstrings_solution_b", "codePath": "./SolutionB.qs" })
+
+Yet another way of getting the desired superposition could be preparing the mix of all possible basis states for all qubits iteratively, keeping the parity on each step.
+
+We start by preparing a superposition of all basis states with parity $0$. To achieve that, we loop through all the qubits except the first one and prepare all of them in equal superposition. In order to maintain the parity of the basis states involved, we use the first qubit and conditionally flip its state using a CNOT gate with each next qubit as the control, so that every time there is a $|1\rangle$ state in the chain, we get back to an even number of $|1\rangle$ states in that basis state.
+
+For example, after the first loop iteration we get the state $\frac12(|00\rangle + |11\rangle)$. After the second iteration we get the state
+
+$$CNOT_{2,0} \frac1{\sqrt2}(|00\rangle + |11\rangle) \otimes \frac1{\sqrt2}(|0\rangle + |1\rangle) = \\\\
```

Comment:
We're doing to stop using \color (see https://github.com/microsoft/qsharp/issues/1435), so let's use \textbf to highlight the parts we want to highlight instead (first and third qubits I believe? color formatting doesn't really work when rendering in the playground or in VS Code notebooks)

Reply:
Noted!